service/controller/clusterapi/v27/resource/ipam/create.go

```diff
@@ -154,24 +160,32 @@ func (r *Resource) getReservedNetworks(ctx context.Context) ([]net.IPNet, error)
 	return reservedSubnets, nil
 }
 
-func (r *Resource) persistAllocatedNetwork(cr v1alpha1.AWSConfig, azs []string) func(ctx context.Context, subnet net.IPNet) error {
+func (r *Resource) persistAllocatedNetwork(cr cmav1alpha1.Cluster, azs []string) func(ctx context.Context, subnet net.IPNet) error {
 	return func(ctx context.Context, subnet net.IPNet) error {
 		return r.splitAndPersistReservedSubnet(ctx, cr, subnet, azs)
 	}
 }
 
-func (r *Resource) splitAndPersistReservedSubnet(ctx context.Context, cr v1alpha1.AWSConfig, subnet net.IPNet, azs []string) error {
-	statusAZs, err := splitSubnetToStatusAZs(subnet, azs)
+func (r *Resource) splitAndPersistReservedSubnet(ctx context.Context, cr cmav1alpha1.Cluster, subnet net.IPNet, azs []string) error {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "updating CR status to persist network allocation and chosen availability zones")
+
+	var providerStatus g8sv1alpha1.AWSClusterStatus
+
+	err := json.Unmarshal(cr.Status.ProviderStatus.Raw, &providerStatus)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", "updating CR status to persist network allocation and chosen availability zones")
+	providerStatus.Provider.Network.CIDR = subnet.String()
+
+	b, err := json.Marshal(providerStatus)
+	if err != nil {
+		return microerror.Mask(err)
+	}
 
-	cr.Status.Cluster.Network.CIDR = subnet.String()
-	cr.Status.AWS.AvailabilityZones = statusAZs
+	cr.Status.ProviderStatus.Raw = b
 
-	_, err = r.g8sClient.ProviderV1alpha1().AWSConfigs(cr.Namespace).UpdateStatus(&cr)
+	_, err = r.cmaClient.ClusterV1alpha1().Clusters(cr.Namespace).UpdateStatus(&cr)
```

Comment:
After this we should actually cancel the reconciliation as we modified the CR. Why aren't doing this yet?

Reply:
We do it here: https://github.com/giantswarm/aws-operator/pull/1632/files#diff-f9b0cf5b3172a168ec6b71853345b7feL79